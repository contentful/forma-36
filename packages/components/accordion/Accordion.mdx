---
title: 'Accordion'
slug: /components/accordion/
github: 'https://github.com/contentful/forma-36/tree/forma-v4/packages/components/accordion'
storybook: 'https://v4-f36-storybook.netlify.app/?path=/story/components-accordion'
typescript: ./src/Accordion.tsx,./src/AccordionItem/AccordionItem.tsx
v4Doc: true
---

import { Heading, Stack, TextLink } from '@contentful/f36-components';
import { ExternalLinkIcon } from '@contentful/f36-icons';
import { Props, PropsHeading } from '@contentful/f36-docs-utils';

Accordions are a good way to deliver a large amount of information. An accordion is a list of headers that after being clicked reveal or hide more content related to them.
The header gives the user a summary of the content and the user decides if they need to see the extended content or not.

## Import

```jsx static=true
import { Accordion } from '@contentful/f36-components';
```

## Examples

The accordion has two variations that define the alignment of the chevron icon: left or right.

### Basic usage

```jsx
<Accordion>
  <Accordion.Item title="What payment methods do you accept?">
    <Text>
      Customers on the Team tier can pay with a credit card (American Express,
      MasterCard or Visa). Enterprise customers have the choice of paying with a
      credit card or wire transfer.
    </Text>
  </Accordion.Item>
</Accordion>
```

### Using it with other components

Other typographic components can be passed as the accordion's title and anything can be used as the accordion's content. For example:

```jsx
<Accordion>
  <Accordion.Item
    title={
      <SectionHeading marginBottom="none">
        What payment methods do you accept?
      </SectionHeading>
    }
  >
    <Stack flexDirection="column">
      <Text marginBottom="spacingS">
        Customers on the Team tier can pay with a credit card (American Express,
        MasterCard or Visa). Enterprise customers have the choice of paying with
        a credit card or wire transfer.
      </Text>
      <Button onClick={() => console.log('clicked!')}>
        Accordionâ€™s button
      </Button>
    </Stack>
  </Accordion.Item>
</Accordion>
```

<PropsHeading storybookPath="/story/components-accordion" />

### Accordion

<Props of="Accordion" />

### Accordion.Item

<Props of="AccordionItem" />

## Content guidelines

- The title should be a short message that summarize the content of the accordion.
- Anything can be passed as the content of the accordion, but often organizing it with Texts and Tables would be enough.
- When using headings, be mindful about the heading levels. The header is a Subheading with `h2` tag.

## Accessibility

- It allows keyboard navigation to open and close the accordions
