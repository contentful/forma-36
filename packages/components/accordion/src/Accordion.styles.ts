import { css, cx } from 'emotion';
import tokens from '@contentful/f36-tokens';

export const getAccordionStyles = ({ className }: { className?: string }) => ({
  accordion: cx(
    css({
      boxSizing: 'border-box',
      padding: '0',
      margin: '0',
      listStyle: 'none',
      '&:first-child': {
        borderTop: `1px solid ${tokens.gray300}`,
      },
    }),
    className,
  ),
});
