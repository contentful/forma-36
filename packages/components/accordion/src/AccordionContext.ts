import React from 'react';
import { Accordion } from '../dist';

export type AccordionContextType = {
  align?: 'start' | 'end';
};

const AccordionContext = React.createContext<AccordionContextType | undefined>(
  undefined,
);

export const useAccordionContext = () => {
  const context = React.useContext(AccordionContext);

  if (context === undefined) {
    throw new Error('component must be rendered within a Accordion component');
  }

  return context;
};

export const AccordionContextProvider = AccordionContext.Provider;
