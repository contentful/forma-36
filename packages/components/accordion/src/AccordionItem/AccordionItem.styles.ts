import { css, cx } from 'emotion';
import tokens from '@contentful/f36-tokens';
import type { CSSObject } from '@emotion/serialize';

const accordionItem = {
  padding: '0',
  margin: '0',
  borderBottom: `1px solid ${tokens.colorElementMid}`,
  '&:first-child': {
    borderTop: `1px solid ${tokens.colorElementMid}`,
  },
} as CSSObject;

export const getAccordionItemStyles = ({
  className,
}: {
  className?: string;
}) => ({
  accordionItem: cx(css(accordionItem), className),
});
