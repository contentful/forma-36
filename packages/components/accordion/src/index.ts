export { Accordion } from './Accordion';
export type { AccordionProps } from './Accordion';
export { AccordionHeader } from './AccordionHeader/AccordionHeader';
export type { AccordionHeaderProps } from './AccordionHeader/AccordionHeader';
export { AccordionItem } from './AccordionItem/AccordionItem';
export type { AccordionItemProps } from './AccordionItem/AccordionItem';
export { AccordionPanel } from './AccordionPanel/AccordionPanel';
export type { AccordionPanelProps } from './AccordionPanel/AccordionPanel';
