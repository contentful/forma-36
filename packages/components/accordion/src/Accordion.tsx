import React from 'react';
import { Box, type CommonProps, type ExpandProps } from '@contentful/f36-core';

import { getAccordionStyles } from './Accordion.styles';

export interface AccordionProps extends CommonProps {
  /**
   * Specify the alignment of the chevron inside the accordion header
   * @default end
   */
  align?: 'start' | 'end';
  /**
   * Child nodes to be rendered in the component
   */
  children?: React.ReactNode;
}

const _Accordion = (
  {
    align = 'end',
    children,
    className,
    testId = 'cf-ui-accordion',
    ...otherProps
  }: ExpandProps<AccordionProps>,
  ref: React.Ref<HTMLUListElement>,
) => {
  const styles = getAccordionStyles({ className });
  return (
    <Box
      as="ul"
      className={styles.accordion}
      testId={testId}
      {...otherProps}
      ref={ref}
    >
      {React.Children.map(children, (child) => {
        if (React.isValidElement(child)) {
          return React.cloneElement(child, {
            align,
          });
        }
        return child;
      })}
    </Box>
  );
};

export const Accordion = React.forwardRef(_Accordion);
