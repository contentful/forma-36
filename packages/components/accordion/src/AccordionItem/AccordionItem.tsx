import React from 'react';
import type { HeadingElement } from '@contentful/f36-typography';
import { useId, Box, useControllableState } from '@contentful/f36-core';

import { AccordionHeader } from '../AccordionHeader/AccordionHeader';
import { AccordionPanel } from '../AccordionPanel/AccordionPanel';
import type { CommonProps, ExpandProps } from '@contentful/f36-core';

import { getAccordionItemStyles } from './AccordionItem.styles';

export interface AccordionItemProps extends CommonProps {
  /**
   * The accordion title
   */
  title?: React.ReactNode;
  /**
   * The heading element that will be used by the AccordionHeader
   */
  titleElement?: HeadingElement;
  /**
   * The children of the AccordionItem are in fact the content of the accordion
   */
  children?: React.ReactNode;
  /**
   * A function to be called when the accordion item is opened
   */
  onExpand?: () => void;
  /**
   * A function to be called when the accordion item is closed
   */
  onCollapse?: () => void;
  /**
   * Specify the alignment of the chevron inside the accordion header
   */
  align?: 'start' | 'end';

  /**
   * By default, the AccordionItem is uncontrolled (manage it's expanded state by itself)
   * But you can make it controlled by providing boolean
   */
  isExpanded?: boolean;
}

const _AccordionItem = (
  {
    title = 'Accordion Title',
    titleElement = 'h2',
    testId = 'cf-ui-accordion-item',
    onExpand,
    onCollapse,
    children,
    align = 'end',
    className,
    isExpanded,
    ...otherProps
  }: ExpandProps<AccordionItemProps>,
  ref: React.Ref<HTMLLIElement>,
) => {
  const styles = getAccordionItemStyles({ className });
  const id = useId();
  const { isOpen, handleOpen, handleClose } = useControllableState({
    isOpen: isExpanded,
    onOpen: onExpand,
    onClose: onCollapse,
  });

  const handleOnClick = () => {
    if (isOpen) {
      handleClose();
    } else {
      handleOpen();
    }
  };

  return (
    <Box
      as="li"
      className={styles.accordionItem}
      testId={`${testId}-${id}`}
      {...otherProps}
      ref={ref}
    >
      <AccordionHeader
        onClick={handleOnClick}
        isExpanded={isOpen}
        element={titleElement}
        ariaId={id}
        align={align}
      >
        {title}
      </AccordionHeader>

      <AccordionPanel ariaId={id} isExpanded={isOpen}>
        {children}
      </AccordionPanel>
    </Box>
  );
};

export const AccordionItem = React.forwardRef(_AccordionItem);
