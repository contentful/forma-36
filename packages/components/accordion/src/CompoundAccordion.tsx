import { Accordion as OriginalAccordion } from './Accordion';
import { AccordionItem } from './AccordionItem/AccordionItem';

type CompoundAccordion = typeof OriginalAccordion & {
  Item: typeof AccordionItem;
};

export const Accordion = OriginalAccordion as CompoundAccordion;
Accordion.Item = AccordionItem;
