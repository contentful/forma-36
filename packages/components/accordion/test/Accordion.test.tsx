import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { axe } from '@/scripts/test/axeHelper';

import { Accordion } from '../src/Accordion';
import { AccordionItem } from '../src/AccordionItem/AccordionItem';

jest.mock('@contentful/f36-core', () => ({
  ...jest.requireActual('@contentful/f36-core'),
  useId: () => {
    return 'id';
  },
}));

it('opens the accordion panel when the header is clicked', () => {
  render(
    <Accordion>
      <AccordionItem title="Accordion Title">Accordion content</AccordionItem>
    </Accordion>,
  );

  const panel = screen.getByLabelText('Accordion Title');

  expect(panel.getAttribute('aria-hidden')).toBe('true');

  userEvent.click(screen.getByText('Accordion Title'));
  expect(panel.getAttribute('aria-hidden')).toBe('false');
});

it('calls onExpand && onCollapse when an accordion item is expanded and collapsed', () => {
  const onExpand = jest.fn();
  const onCollapse = jest.fn();
  render(
    <Accordion>
      <AccordionItem
        title="Accordion Title"
        onExpand={onExpand}
        onCollapse={onCollapse}
      >
        Accordion content
      </AccordionItem>
    </Accordion>,
  );

  userEvent.click(screen.getByText('Accordion Title'));
  expect(onExpand).toHaveBeenCalledTimes(1);
  expect(onCollapse).toHaveBeenCalledTimes(0);

  userEvent.click(screen.getByText('Accordion Title'));
  expect(onExpand).toHaveBeenCalledTimes(1);
  expect(onCollapse).toHaveBeenCalledTimes(1);
});

it('has no a11y issues', async () => {
  const { container } = render(
    <Accordion>
      <AccordionItem title="Accordion Title">Accordion content</AccordionItem>
    </Accordion>,
  );
  const results = await axe(container);

  expect(results).toHaveNoViolations();
});
