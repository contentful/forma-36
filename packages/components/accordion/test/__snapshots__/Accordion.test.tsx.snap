// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
.emotion-6 {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  list-style: none;
}

.emotion-6:first-child {
  border-top: 1px solid #CFD9E0;
}

.emotion-4 {
  box-sizing: border-box;
  overflow: hidden;
  height: 0;
  -webkit-transition: height 0.2s ease-in-out,padding 0.2s ease-in-out;
  transition: height 0.2s ease-in-out,padding 0.2s ease-in-out;
}

.emotion-3 {
  width: 100%;
  padding: 0.5rem 1rem 1rem;
}

.emotion-5 {
  padding: 0;
  margin: 0;
  border-bottom: 1px solid #CFD9E0;
}

.emotion-5:first-child {
  border-top: 1px solid #CFD9E0;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  padding: 1rem;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5;
  color: #1B273A;
  width: 100%;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-1:hover {
  background-color: #F7F9FA;
}

.emotion-1:focus {
  outline: none;
  background-color: #F7F9FA;
}

.emotion-2 {
  display: block;
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-weight: 600;
  color: #111B2B;
  font-size: 1.25rem;
  line-height: 2rem;
  text-rendering: optimizeLegibility;
  font-size: 1rem;
  line-height: 1.5rem;
}

.emotion-0 {
  display: inline-block;
  fill: #111B2B;
  height: 18px;
  width: auto;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 0.2s ease-in-out;
  -webkit-transition: transform 0.2s ease-in-out;
  transition: transform 0.2s ease-in-out;
}

<ul
  class="emotion-6"
  data-test-id="cf-ui-accordion"
>
  <li
    class="emotion-5"
    data-test-id="cf-ui-accordion-item-id"
  >
    <h2
      class="emotion-2"
      data-test-id="cf-ui-subheading"
    >
      <button
        aria-controls="accordion-panel--id"
        aria-expanded="false"
        class="emotion-1"
        id="accordion--id"
        type="button"
      >
        <svg
          class="emotion-0"
          data-test-id="cf-ui-icon"
          height="1em"
          viewBox="0 0 9 18"
          width="1em"
        >
          <path
            d="M1.1 5.9l3.4 3.4 3.4-3.4 1.1 1-4.5 4.5L0 6.9l1.1-1z"
          />
          <path
            d="M0 0h9v18H0V0z"
            fill="none"
          />
        </svg>
        Accordion Title
      </button>
    </h2>
    <div
      aria-hidden="true"
      aria-labelledby="accordion--id"
      class="emotion-4"
      id="accordion-panel--id"
      role="region"
    >
      <div
        class="emotion-3"
      >
        Accordion content
      </div>
    </div>
  </li>
</ul>
`;

exports[`renders the component with an additional class name 1`] = `
.emotion-6 {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  list-style: none;
}

.emotion-6:first-child {
  border-top: 1px solid #CFD9E0;
}

.emotion-4 {
  box-sizing: border-box;
  overflow: hidden;
  height: 0;
  -webkit-transition: height 0.2s ease-in-out,padding 0.2s ease-in-out;
  transition: height 0.2s ease-in-out,padding 0.2s ease-in-out;
}

.emotion-3 {
  width: 100%;
  padding: 0.5rem 1rem 1rem;
}

.emotion-5 {
  padding: 0;
  margin: 0;
  border-bottom: 1px solid #CFD9E0;
}

.emotion-5:first-child {
  border-top: 1px solid #CFD9E0;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  padding: 1rem;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5;
  color: #1B273A;
  width: 100%;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-1:hover {
  background-color: #F7F9FA;
}

.emotion-1:focus {
  outline: none;
  background-color: #F7F9FA;
}

.emotion-2 {
  display: block;
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-weight: 600;
  color: #111B2B;
  font-size: 1.25rem;
  line-height: 2rem;
  text-rendering: optimizeLegibility;
  font-size: 1rem;
  line-height: 1.5rem;
}

.emotion-0 {
  display: inline-block;
  fill: #111B2B;
  height: 18px;
  width: auto;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 0.2s ease-in-out;
  -webkit-transition: transform 0.2s ease-in-out;
  transition: transform 0.2s ease-in-out;
}

<ul
  class="emotion-6"
  data-test-id="cf-ui-accordion"
>
  <li
    class="emotion-5"
    data-test-id="cf-ui-accordion-item-id"
  >
    <h2
      class="emotion-2"
      data-test-id="cf-ui-subheading"
    >
      <button
        aria-controls="accordion-panel--id"
        aria-expanded="false"
        class="emotion-1"
        id="accordion--id"
        type="button"
      >
        <svg
          class="emotion-0"
          data-test-id="cf-ui-icon"
          height="1em"
          viewBox="0 0 9 18"
          width="1em"
        >
          <path
            d="M1.1 5.9l3.4 3.4 3.4-3.4 1.1 1-4.5 4.5L0 6.9l1.1-1z"
          />
          <path
            d="M0 0h9v18H0V0z"
            fill="none"
          />
        </svg>
        Accordion Title
      </button>
    </h2>
    <div
      aria-hidden="true"
      aria-labelledby="accordion--id"
      class="emotion-4"
      id="accordion-panel--id"
      role="region"
    >
      <div
        class="emotion-3"
      >
        Accordion content
      </div>
    </div>
  </li>
</ul>
`;

exports[`renders the component with chevron on the left 1`] = `
.emotion-6 {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
  list-style: none;
}

.emotion-6:first-child {
  border-top: 1px solid #CFD9E0;
}

.emotion-4 {
  box-sizing: border-box;
  overflow: hidden;
  height: 0;
  -webkit-transition: height 0.2s ease-in-out,padding 0.2s ease-in-out;
  transition: height 0.2s ease-in-out,padding 0.2s ease-in-out;
}

.emotion-3 {
  width: 100%;
  padding: 0.5rem 1rem 1rem;
}

.emotion-5 {
  padding: 0;
  margin: 0;
  border-bottom: 1px solid #CFD9E0;
}

.emotion-5:first-child {
  border-top: 1px solid #CFD9E0;
}

.emotion-2 {
  display: block;
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-weight: 600;
  color: #111B2B;
  font-size: 1.25rem;
  line-height: 2rem;
  text-rendering: optimizeLegibility;
  font-size: 1rem;
  line-height: 1.5rem;
}

.emotion-0 {
  display: inline-block;
  fill: #111B2B;
  height: 18px;
  width: auto;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: -webkit-transform 0.2s ease-in-out;
  -webkit-transition: transform 0.2s ease-in-out;
  transition: transform 0.2s ease-in-out;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0;
  padding: 1rem;
  background-color: transparent;
  font-family: -apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5;
  color: #1B273A;
  width: 100%;
  cursor: pointer;
  -webkit-transition: background-color 0.2s ease-in-out;
  transition: background-color 0.2s ease-in-out;
  min-width: 9px;
  margin-right: 0.5rem;
}

.emotion-1:hover {
  background-color: #F7F9FA;
}

.emotion-1:focus {
  outline: none;
  background-color: #F7F9FA;
}

<ul
  class="emotion-6"
  data-test-id="cf-ui-accordion"
>
  <li
    class="emotion-5"
    data-test-id="cf-ui-accordion-item-id"
  >
    <h2
      class="emotion-2"
      data-test-id="cf-ui-subheading"
    >
      <button
        aria-controls="accordion-panel--id"
        aria-expanded="false"
        class="emotion-1"
        id="accordion--id"
        type="button"
      >
        <svg
          class="emotion-0"
          data-test-id="cf-ui-icon"
          height="1em"
          viewBox="0 0 9 18"
          width="1em"
        >
          <path
            d="M1.1 5.9l3.4 3.4 3.4-3.4 1.1 1-4.5 4.5L0 6.9l1.1-1z"
          />
          <path
            d="M0 0h9v18H0V0z"
            fill="none"
          />
        </svg>
        Accordion Title
      </button>
    </h2>
    <div
      aria-hidden="true"
      aria-labelledby="accordion--id"
      class="emotion-4"
      id="accordion-panel--id"
      role="region"
    >
      <div
        class="emotion-3"
      >
        Accordion content
      </div>
    </div>
  </li>
</ul>
`;
